docs/pages/tutorials/embedded-mode-with-nextjs-frontend.mdx

```diff
@@ -0,0 +1,271 @@
+
+
+## Add Keystone to your project
+
+Now that we have the NextJS starter with static files, let‘s embed Keystone into the app to blend file-based content with content you can edit using Keystone’s intuitive Admin UI.
+
+### Install dependencies
+
+Add the following Keystone dependencies to the project:
+
+```bash
+yarn add @keystone-next/keystone @keystone-next/admin-ui @keystone-next/fields
+```
+
+### Update .gitignore
+
+Add the `.keystone` directory to your `.gitignore` file. The contents of `.keystone` are dynamically generated at build time and you’ll never have to change them.
+
+```bash
+
+# in your .gitignore
+.keystone
+```
+
+### Add Keystone to NextJS config
+
+Add a `next.config.js` file to the project root with the following:
+
+```tsx
+
+// next.config.js
+
+const { withKeystone } = require('@keystone-next/keystone/next');
+
+module.exports = withKeystone();
+```
+
+This is where the magic happens – the `withKeystone` constant lets NextJS encapsulate Keystone in its script runtime, but Keystone still operates independently of the NextJS frontend.✨
+
+### Create a content List in Keystone config
+
+Now add a `keystone.ts` [configuration file](/apis/config) to the project root with a simple `Post` [list](/apis/schema) type. This will give us the ability to create and edit blog records in Keystone’s Admin UI:
+
+```tsx
+// keystone.ts
+
+import { config, list } from '@keystone-next/keystone/schema';
+import { text } from '@keystone-next/fields';
+
+const Post = list({
+  fields: {
+    title: text({ isRequired: true }),
+    slug: text(),
+    content: text(),
+  },
+});
+
+export default config({
+  db: { adapter: 'prisma_sqlite', url: 'file:./app.db' },
+  experimental: {
+    generateNextGraphqlAPI: true,
+    generateNodeAPI: true,
+  },
+  lists: { Post },
+});
+```
+
+### Update package scripts
+
+Finally, make a small change to the `scripts` object in `package.json` to include Keystone‘s `postinstall`:
+
+```json
+	// within package.json
+
+  "scripts": {
+    "postinstall": "keystone-next postinstall",
+    "dev": "next dev",
+    "start": "next start",
+    "build": "next build"
+  },
+```
+
+Running `yarn dev` again will do the following:
+
+- Provision a GraphQL schema based on the configuration of `keystone.ts`.
+- Build a [Prisma.io](https://www.prisma.io/) schema (which Keystone uses to create tables in the database).
+- Serve Keystone’s Admin UI at [http://localhost:8000](http://localhost:8000).
+- Serve the NextJS frontend at [http://localhost:3000](http://localhost:3000).
+- Add a `postinstall` script that ensures everything works if we install dependencies in the future.
+
+![Local server instances of the NExtJS frontend, and Keystone Admin UI](/assets/tutorials/next-lite/localhost-next-keystone.gif)
+
+Go ahead and add two post entries using your Admin UI, ensuring the Slug field `uses-hyphens-and-lowercase-chars` for permalinks.
```

Comment:
```suggestion
Go ahead and add two post entries using your Admin UI, ensuring you only use `hyphens-and-lowercase-chars` in the slug field for permalinks.
```

Could be a great place for an info box linking out to how to set up validation for text fields, when we get to that 😉

Reply:
agreed. there's a great use case for a link to a specific tuts on validation, and slugifying things down the track